File: infrastructure/src/main/java/org/haas/infrastructure/persistence/movie/entity/MovieEntity.java
package org.haas.infrastructure.persistence.movie.entity;

import jakarta.persistence.*;
import lombok.*;
import org.haas.core.domain.genre.Genre;
import org.haas.core.domain.movie.Movie;
import org.haas.core.domain.movie.MovieStatus;
import org.haas.core.domain.screening.Screening;
import org.haas.infrastructure.persistence.screening.entity.ScreeningEntity;

import java.time.LocalDate;
import java.util.List;

@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Getter
@Table(name = "movies")
@Entity
public class MovieEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private Genre genre;

    private int runningTime;

    private String director;

    private String filmRating;

    private LocalDate releasedAt;

    private String thumbnailUrl;

    @Enumerated(EnumType.STRING)
    private MovieStatus movieStatus;

    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id")
    private List<ScreeningEntity> screenings;


}

File: data/src/main/kotlin/kr/spartacodingclub/cinema/data/entity/MovieEntity.kt
package kr.spartacodingclub.cinema.data.entity

import jakarta.persistence.*
import kr.spartacodingclub.cinema.core.domain.Genre
import kr.spartacodingclub.cinema.core.domain.Movie
import kr.spartacodingclub.cinema.core.domain.MovieRating
import java.time.LocalDate

@Entity
@Table(name = "movie")
class MovieEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,

    val title: String,

    @Enumerated(EnumType.STRING)
    val rating: MovieRating,

    val releaseDate: LocalDate,
    val thumbnailUrl: String,
    val runningTime: Int,

    @Enumerated(EnumType.STRING)
    val genre: Genre
) : BaseEntity() {
    fun toDomain() = Movie(
        id = id,
        title = title,
        rating = rating,
        releaseDate = releaseDate,
        thumbnailUrl = thumbnailUrl,
        runningTime = runningTime,
        genre = genre
    )
}
